code/common/dataaccess.q

```diff
@@ -26,6 +26,7 @@ metainfo:([tablename:`$()]partfield:`$();metas:();proctype:`$());
 //-   - load config for checking input parameters
 //-   - write meta info for tables in current process to .dataaccess.metainfo
 init:{[tablepropertiespath]
+  if[.proc.proctype = `gateway; .aqrest.execute::{[req;props] @[value;req;{(neg .z.w)(.gw.formatresponse[0b;0b;"error: ",x])}]}];
```

Comment:
I don't think we want to hardcode based on processtype. We shouldn't require gateways to be off type "gateway" as it is too restrictive

I think what we need to do instead is add a new configuration parameter to the gateway config, which if true resets the definition of .aqrest.execute to the new definition (this will need commented and documented because it will be tricky to find). This will give us more flexibility (e.g. it should be possible to run the GW with DA loaded but not resetting the definition of .aqrest.execute as well)

Reply:
Changed the condition to check the value (true/false) of .aqrest.loadexecute instead of the process type. The default value is set in /config/settings/gateway.q.